src/snowcli/cli/nativeapp/run_processor.py

```diff
@@ -227,9 +228,16 @@ def get_existing_version_info(self, version: str) -> Optional[dict]:
         """
         with self.use_role(self.package_role):
             show_obj_query = f"show versions like '{unquote_identifier(version)}' in application package {self.package_name}"
```

Comment:
Feels like we should have a utility to do this already. If we don't, can we create one?

Reply:
Not sure I follow. This query is part of `get_existing_version_info`, and this is the only query executed in this method.

Reply:
I mean the "get all versions" part of get_existing_version_info. The first part of the utility function seems to just try to find all versions, and then we look for the desired one. Felt like maybe that's something that should be factored out?

Reply:
For instance, right above there's "get_all_existing_versions", but if the app package doesn't exit, I think that function has a different behaviour w.r.t. raised exceptions. Seemed odd is all.

Reply:
Oh right, I did try to unify the two: `get_all_existing_versions` and `get_existing_version_info` by only performing the filtering on the returned results of `get_all_existing_versions` in `get_existing_version_info`. However, the way that the results are returned in `SnowflakeCursor` is an odd one to handle. Here's why:

If I try to dictate the `_execute_query` method within `get_all_existing_versions` to use `DictCursor` in order to filter on version name in `get_existing_version_info`, the `_execute_query` ends up wiping the values in the returned results entirely, this was discovered when I ran the integration tests for the PR that introduced the change: https://github.com/Snowflake-Labs/snowcli/actions/runs/7416652530/job/20181962128

This is why `get_existing_version_info` performs `show versions like <version> ... ` whereas `get_all_existing_versions` does a `show versions in ... `because the latter does not require filtering. 

In addition, there are a few other methods that also execute a singular query, and perform a filter. Some examples are: `NativeAppManager.get_existing_app_info`, `NativeAppManager.get_existing_app_pkg_info` and `NativeAppVersionCreateProcessor.add_new_patch_to_version`. That would make `get_existing_version_info` consistent with how these are written to. 

Re: _right above there's "get_all_existing_versions", but if the app package doesn't exit, I think that function has a different behaviour w.r.t. raised exceptions_ - I feel like this is a separate point about making the exceptions behavior consistent between the two methods, at the very least. 

Happy to chat about this on slack.

Reply:
Thanks for the detailed explanation. Overall I feel there's a bit of a lack of abstraction in our code, but that's a minor point here. You're right, what jumped out at me is that depending on the utility method, the behaviour seems slightly inconsistent, at least outside of the happy path. To me these two things are somewhat related, but that might be something to discuss as a team in the near future instead. Don't let this block this PR.